src/libstd/backtrace.rs

```diff
@@ -159,6 +159,63 @@ enum BytesOrWide {
     Wide(Vec<u16>),
 }
 
+impl fmt::Debug for Backtrace {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut capture = match &self.inner {
+            Inner::Unsupported => return fmt.write_str("unsupported backtrace"),
+            Inner::Disabled => return fmt.write_str("disabled backtrace"),
+            Inner::Captured(c) => c.lock().unwrap(),
+        };
+        capture.resolve();
+
+        write!(fmt, "Backtrace ")?;
+
+        let mut dbg = fmt.debug_list();
+
+        for frame in &capture.frames {
+            dbg.entries(&frame.symbols);
+        }
+
+        dbg.finish()
+    }
+}
+
+impl fmt::Debug for BacktraceSymbol {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut dbg = fmt.debug_struct("");
+
+        if let Some(fn_name) = self.name.as_ref().map(|b| backtrace::SymbolName::new(b)) {
+            dbg.field("fn", &format_args!("\"{}\"", fn_name));
+        } else {
+            dbg.field("fn", &"<unknown>");
+        }
+
+        if let Some(fname) = self.filename.as_ref() {
+            dbg.field("file", fname);
+        }
+
+        if let Some(line) = self.lineno.as_ref() {
+            dbg.field("line", line);
+        }
+
+        dbg.finish()
```

Comment:
I think it would be nicer to format each symbol on one line even in `{:#?}` mode. Backtraces will normally have 15+ frames and shouldn't take this many lines in a debug printed error. Compare:

```console
Error {
    message: "the thing went wrong",
    backtrace: Backtrace [
        { fn: "std::backtrace::Backtrace::create::ha19bbd9f10287f80" },
        { fn: "std::backtrace::Backtrace::force_capture::h5e8065a48ff91123" },
        { fn: "std::backtrace::tests::debug_backtrace_fmt::h0557493242cdd27b" },
        { fn: "<alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::h94c20d903b9ee892" },
        { fn: "__rust_maybe_catch_panic" },
        { fn: "std::panicking::try::hc8e0ebad5360bb9e" },
        { fn: "test::run_test_in_process::h23ee0e24f420835e" },
        { fn: "std::sys_common::backtrace::__rust_begin_short_backtrace::h66d96ab05a49a4cf" },
        { fn: "std::panicking::try::do_call::he44bd66b51853864" },
        { fn: "__rust_maybe_catch_panic" },
        { fn: "std::panicking::try::h9e6571690ab8a505" },
        { fn: "core::ops::function::FnOnce::call_once{{vtable.shim}}::hb78017e7280200c9" },
        { fn: "<alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::h6caad6f61b1e39b2" },
        { fn: "<alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::hc9d5f44518fec1bf" },
        { fn: "std::sys_common::thread::start_thread::h44e11c2292c6e0da" },
        { fn: "std::sys::unix::thread::Thread::new::thread_start::h1534b513264a5d02" },
        { fn: "start_thread" },
        { fn: "__clone" },
    ],
}
```

vs:

```console
Error {
    message: "the thing went wrong",
    backtrace: Backtrace [
        {
            fn: "std::backtrace::Backtrace::create::ha19bbd9f10287f80",
        },
        {
            fn: "std::backtrace::Backtrace::force_capture::h5e8065a48ff91123",
        },
        {
            fn: "std::backtrace::tests::debug_backtrace_fmt::h0557493242cdd27b",
        },
        {
            fn: "<alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::h94c20d903b9ee892",
        },
        {
            fn: "__rust_maybe_catch_panic",
        },
        {
            fn: "std::panicking::try::hc8e0ebad5360bb9e",
        },
        {
            fn: "test::run_test_in_process::h23ee0e24f420835e",
        },
        {
            fn: "std::sys_common::backtrace::__rust_begin_short_backtrace::h66d96ab05a49a4cf",
        },
        {
            fn: "std::panicking::try::do_call::he44bd66b51853864",
        },
        {
            fn: "__rust_maybe_catch_panic",
        },
        {
            fn: "std::panicking::try::h9e6571690ab8a505",
        },
        {
            fn: "core::ops::function::FnOnce::call_once{{vtable.shim}}::hb78017e7280200c9",
        },
        {
            fn: "<alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::h6caad6f61b1e39b2",
        },
        {
            fn: "<alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::hc9d5f44518fec1bf",
        },
        {
            fn: "std::sys_common::thread::start_thread::h44e11c2292c6e0da",
        },
        {
            fn: "std::sys::unix::thread::Thread::new::thread_start::h1534b513264a5d02",
        },
        {
            fn: "start_thread",
        },
        {
            fn: "__clone",
        },
    ],
}
```

Reply:
I have no idea how to do this via the `debug_map` api so I'm just gonna use manual formatting via `write!` for the BacktraceFrame's debug impl to try to do this.